### ISSUE TITLE
How can i make a radiogroup indeterminate?

### ISSUE THREAD

**Issue by jimmywarting** [react-toolbox/react-toolbox#1788]
Got a weird request from a customer.
he basically suggested that if you click on a checked radio button it would deselect

kinda like this:
https://jsfiddle.net/tuanvuf4/b5w9cd3L/

**Comment by rubenmoya** [comment]
You have a working example in the playground: http://react-toolbox.io/#/components/radio_group basically give a name to the `RadioGroup`, a value to each `RadioButton` and then pass the value you want to select to the `RadioGroup` component

**Comment by 3af** [comment]
@rubenmoya Can you please share the code? I wasn't able to do that.

The problem seems to be that the [onChange handler](https://github.com/react-toolbox/react-toolbox/blob/dev/components/radio/RadioButton.js#L44) is not called when the button is already checked (which makes sense because there is no change). This means you don't know when user clicks on the currently checked option and therefore you can't un-check it (set `value` of `RadioGroup` to `null` which would solve the problem).

So I think the answer to the question is: "You can't". Please correct me if I'm wrong.

One solution would be to add new prop, `allowIndeterminate` to `RadioGroup`, and then check it in the RadioButton#handleClick method. But I'm not sure whether it doesn't go against the philosophy of RadioGroup. Maybe you should use checkboxes @jimmywarting.

**Comment by jimmywarting** [comment]
But i need to add a click handler to the radio button so i can pass the (undefined) value that i want to RadioGroup?

**Comment by jimmywarting** [comment]
@3af you are correct, it's currently not possible onChange won't be called if clicking it twice

`allowIndeterminate` would be a nice option.
at the very least a onClick handler so i can do it myself

Thought about using checkboxes but no...

**Comment by rubenmoya** [comment]
@3af You're right, I misunderstood the issue. Right now it's not possible to do it. I think the onClick prop would be a better option than `allowIndeterminate` since it's more flexible.

**Comment by 3af** [comment]
I agree, the `onClick` is better solution.

**Comment by junaidn** [comment]
Can I try to attempt it ?

**Comment by shreeharshshinde** [comment]
I've implemented a workaround using a custom onClick handler to toggle the selected radio off when it's clicked again. This manually sets the value to null, effectively making the radio group behave like an indeterminate checkbox. Since onChange doesn't fire for already-selected radios, this pattern uses onClick for full control. While it's a non-standard behavior, it's useful for special UX cases where deselection is needed.

**Comment by shreeharshshinde** [comment]
### Proposed Enhancement for Deselecting Radio Buttons

I’ve implemented a clean JavaScript workaround that allows users to deselect a selected radio button by clicking it again — essentially mimicking an "indeterminate" state within a radio group.

🔍 How it works:

- On `mousedown`, we record whether the radio was already checked.

- On `click`, if it was previously checked, we manually set `.checked = false.`

- A `change` event is manually triggered to notify listeners.

This enables behavior similar to checkboxes, but with radios, and works even across multiple groups.

This could be considered a UX enhancement where selection is optional, or useful in cases where a radio acts as a toggle.

If this approach aligns with the project’s direction, I’d love to contribute a PR — maybe via a prop like `allowIndeterminate`, or a hook into `onClick`.

**Comment by shreeharshshinde** [comment]
```// Enhancement: Make radio buttons deselectable on second click
document.querySelectorAll('input[type=radio]').forEach(radio => {
    radio.addEventListener('mousedown', function () {
        // Save whether it was checked before the click
        radio.dataset.wasChecked = radio.checked ? "true" : "false";
    });

    radio.addEventListener('click', function () {
        // If it was checked already, uncheck it and trigger change
        if (radio.dataset.wasChecked === "true") {
            radio.checked = false;
            radio.dispatchEvent(new Event('change', { bubbles: true }));
        }
    });
});
```
